Python/pythonrun.c

```diff
@@ -1275,7 +1275,10 @@ run_eval_code_obj(PyThreadState *tstate, PyCodeObject *co, PyObject *globals, Py
     _PyRuntime.signals.unhandled_keyboard_interrupt = 0;
 
     /* Set globals['__builtins__'] if it doesn't exist */
-    if (globals != NULL) {
+    if (globals == NULL) {
+        PyErr_SetString(PyExc_RuntimeError, "globals are NULL");
```

Comment:
Would this error message be helpful for the user if this branch is triggered?

Reply:
I don't know, do you have any suggestions how to make it more clear?
As i understand it is module dictionary and it shouldn't be NULL.
If it is, then something goes completly wrong.
Or you mean to make message longer, include some explanation of what happend?

Reply:
I'm back with some study. Made another analysis of calls, everything seems to be ok inside interpreter, but if someone is using C API then error message may help to figure out what's happening.
Below is my NULL pointer C API example:

```C
#include <stdio.h>
#include <Python.h>

int main() {

  PyObject *pdict, *pResult;
  char *buff = "script.py";
  int result = 0;

  Py_Initialize();

  /*
  pdict = PyDict_New();
  pResult = PyRun_FileEx(fopen(buff, "r"), buff, Py_file_input, pdict, pdict, 1);
  */
  pResult = PyRun_FileEx(fopen(buff, "r"), buff, Py_file_input, NULL, NULL, 1);

  if (pResult == NULL) {

    PyObject *ptype, *pvalue, *ptraceback;

    PyErr_Fetch(&ptype, &pvalue, &ptraceback);

    printf("Exception while calling script\n");
    if (ptype != NULL) {
      printf("Type: %s\n", PyUnicode_AsUTF8(PyObject_Str(ptype)));
    }
    if (pvalue != NULL) {
      printf("Error: %s\n", PyUnicode_AsUTF8(PyObject_Str(pvalue)));
    }
    if (ptraceback != NULL) {
      printf("Trace: %s\n", PyUnicode_AsUTF8(PyObject_Str(ptraceback)));
    }

    result = -1;

  }
  else {

    printf("pResult != NULL\n");
    Py_DECREF(pResult);

  }

  Py_DECREF(pdict);
  Py_Finalize();

  return result;
}
```

It runs sample script:

```python
def myprint(s: str):

    print(s)

if __name__ == '__main__':

    myprint('run test print')

else:

    print(__name__)
```

When i run compiled binary with standart build (without ./configure --with-pydebug) installed in /usr/local, i've got:

 ```sh
 $ ./test_pyrun_fileex.sh
Segmentation fault
```

My patch will give option to handle error:

```sh
 $ ./test_pyrun_fileex.sh
Exception while calling script
Type: <class 'RuntimeError'>
Error: run_eval_code_obj(): globals=NULL
```

Reply:
Great, can you a regression test to one of the `Modules/_testcapi/*.c` files, or just directly to `Modules/_testcapimodule.c`?